navext/navext-runtime/src/main/java/com/foursoft/harness/navext/runtime/io/write/XMLWriter.java

```diff
@@ -154,14 +155,30 @@ private void write(final T container, final XMLMeta meta, final Writer output) {
 
             final MarshallerListener marshallerListener = new MarshallerListener();
             marshaller.setListener(marshallerListener);
-            meta.getComments().ifPresent(c -> marshallerListener.addListener(new CommentAdderListener(xsw, c)));
+
+            final XMLMeta safeXmlMeta = new XMLMeta();
             meta.getProcessingInstructions()
+                    .ifPresent(safeXmlMeta::setProcessingInstructions);
+            meta.getComments()
+                    .ifPresent(comments -> {
+                        // Comments need a special handling.
+                        final Comments sanitizedComments = new Comments();
+                        comments.getCommentMap().forEach((k, v) -> {
+                            final String sanitizedComment = v.replace("--", "- -");
+                            sanitizedComments.put(k, sanitizedComment);
+                        });
+                        safeXmlMeta.setComments(sanitizedComments);
+                    });
```

Comment:
@RainerGanss: Not sure if it wouldn't be better to adjust the comments when putting them in the map so there won't be an "unsafe" `XMLMeta` object in first place. What do you think?

Reply:
Comments#put(Object key, String comment) was honestly the only method I expected a change for this PR. Please move the sanitizer code there. For any other cases the catch block is fine.